
- remove fragile installed/not-installed detection
- support --cc and --cpp arguments (from CVS HEAD)

--- apr-0.9.4/apr-config.in.config
+++ apr-0.9.4/apr-config.in
@@ -75,7 +75,6 @@
 LIBS="@EXTRA_LIBS@"
 EXTRA_INCLUDES="@EXTRA_INCLUDES@"
 SHLIBPATH_VAR="@shlibpath_var@"
-APR_SOURCE_DIR="@apr_srcdir@"
 APR_SO_EXT="@so_ext@"
 APR_LIB_TARGET="@export_lib_target@"
 APR_LIBNAME="@APR_LIBNAME@"
@@ -123,37 +122,7 @@
     exit 1
 fi
 
-thisdir="`dirname $0`"
-thisdir="`cd $thisdir && pwd`"
-if test -d $bindir; then
-  tmpbindir="`cd $bindir && pwd`"
-else
-  tmpbindir=""
-fi
-# If we have the realpath program, use it to resolve symlinks
-# Otherwise, being in a symlinked dir may result in incorrect output.
-if test -x "`which realpath 2>/dev/null`"; then
-  thisdir="`realpath $thisdir`"
-  if test -d "$APR_SOURCE_DIR"; then
-    APR_SOURCE_DIR="`realpath $APR_SOURCE_DIR`"
-  fi
-  if test -n "$tmpbindir"; then
-    tmpbindir="`realpath $tmpbindir`"
-  fi
-fi
-if test "$tmpbindir" = "$thisdir"; then
-  location=installed
-elif test "$APR_SOURCE_DIR" = "$thisdir"; then
-  location=source
-else
-  location=build
-fi
-
-if test "$location" = "installed"; then
-    LA_FILE="$libdir/lib${APR_LIBNAME}.la"
-else
-    LA_FILE="$thisdir/lib${APR_LIBNAME}.la"
-fi
+LA_FILE="$libdir/lib${APR_LIBNAME}.la"
 
 flags=""
 
@@ -178,15 +147,15 @@
     exit 0
     ;;
     --includedir)
-    if test "$location" = "installed"; then
-        flags="$includedir"
-    elif test "$location" = "source"; then
-        flags="$APR_SOURCE_DIR/include"
-    else
-        # this is for VPATH builds
-        flags="$thisdir/include $APR_SOURCE_DIR/include"
-    fi
-    echo $flags
+    echo $includedir
+    exit 0
+    ;;
+    --cc)
+    echo $CC
+    exit 0
+    ;;
+    --cpp)
+    echo $CPP
     exit 0
     ;;
     --cc)
@@ -210,28 +179,14 @@
     flags="$flags $LDFLAGS"
     ;;
     --includes)
-    if test "$location" = "installed"; then
-        flags="$flags -I$includedir $EXTRA_INCLUDES"
-    elif test "$location" = "source"; then
-        flags="$flags -I$APR_SOURCE_DIR/include $EXTRA_INCLUDES"
-    else
-        # this is for VPATH builds
-        flags="$flags -I$thisdir/include -I$APR_SOURCE_DIR/include $EXTRA_INCLUDES"
-    fi
+    flags="$flags -I$includedir $EXTRA_INCLUDES"
     ;;
     --srcdir)
-    echo $APR_SOURCE_DIR
+    echo ${libdir}/apr
     exit 0
     ;;
     --installbuilddir)
-    if test "$location" = "installed"; then
-        echo "${installbuilddir}"
-    elif test "$location" = "source"; then
-        echo "$APR_SOURCE_DIR/build"
-    else
-        # this is for VPATH builds
-        echo "$thisdir/build"
-    fi
+    echo "${installbuilddir}"
     exit 0
     ;;
     --version)
@@ -239,35 +194,17 @@
     exit 0
     ;;
     --link-ld)
-    if test "$location" = "installed"; then
-        ### avoid using -L if libdir is a "standard" location like /usr/lib
-        flags="$flags -L$libdir -l${APR_LIBNAME}"
-    else
-        flags="$flags -L$thisdir -l${APR_LIBNAME}"
-    fi
+    flags="$flags -l${APR_LIBNAME}"
     ;;
     --link-libtool)
-    # If the LA_FILE exists where we think it should be, use it.  If we're
-    # installed and the LA_FILE does not exist, assume to use -L/-l
-    # (the LA_FILE may not have been installed).  If we're building ourselves,
-    # we'll assume that at some point the .la file be created.
-    if test -f "$LA_FILE"; then
-        flags="$flags $LA_FILE"
-    elif test "$location" = "installed"; then
-        ### avoid using -L if libdir is a "standard" location like /usr/lib
-        flags="$flags -L$libdir -l${APR_LIBNAME}"
-    else
-        flags="$flags $LA_FILE"
-    fi
+    flags="$flags $LA_FILE"
     ;;
     --shlib-path-var)
     echo "$SHLIBPATH_VAR"
     exit 0
     ;;
     --apr-la-file)
-    if test -f "$LA_FILE"; then
-        flags="$flags $LA_FILE"
-    fi
+    flags="$flags $LA_FILE"
     ;;
     --apr-so-ext)
     echo "$APR_SO_EXT"
@@ -278,11 +215,7 @@
     exit 0
     ;;
     --apr-libtool)
-    if test "$location" = "installed"; then
-        echo "${installbuilddir}/libtool"
-    else
-        echo "$thisdir/libtool"
-    fi
+    echo "${SHELL} ${installbuilddir}/libtool"
     exit 0
     ;;
     --help)
